trinity/protocol/bcc/servers.py

```diff
@@ -0,0 +1,126 @@
+import itertools
+from typing import (
+    cast,
+    Iterable,
+    Set,
+    Type,
+)
+
+from eth_typing import (
+    Hash32,
+)
+
+from eth_utils import (
+    to_tuple,
+)
+from eth_utils.toolz import (
+    cons,
+    sliding_window,
+    take,
+)
+
+from cancel_token import CancelToken
+
+from p2p import protocol
+from p2p.peer import BasePeer
+from p2p.protocol import Command
+
+from eth.exceptions import BlockNotFound
+from eth.beacon.db.chain import BaseBeaconChainDB
+from eth.beacon.types.blocks import BaseBeaconBlock
+
+from trinity.protocol.common.servers import BaseRequestServer
+from trinity.protocol.bcc.commands import (
+    GetBeaconBlocks,
+    GetBeaconBlocksMessage,
+)
+from trinity.protocol.bcc.peer import (
+    BCCPeer,
+    BCCPeerPool,
+)
+
+
+class BCCRequestServer(BaseRequestServer):
+    subscription_msg_types: Set[Type[Command]] = {
+        GetBeaconBlocks,
+    }
+
+    def __init__(self,
+                 db: BaseBeaconChainDB,
+                 peer_pool: BCCPeerPool,
+                 token: CancelToken = None) -> None:
+        super().__init__(peer_pool, token)
+        self.db = db
+
+    async def _handle_msg(self, base_peer: BasePeer, cmd: Command,
+                          msg: protocol._DecodedMsgType) -> None:
+        peer = cast(BCCPeer, base_peer)
+
+        if isinstance(cmd, GetBeaconBlocks):
+            await self._handle_get_beacon_blocks(peer, cast(GetBeaconBlocksMessage, msg))
```

Comment:
I'm *much* more comfortable with this type of `cast` than with the other so :+1: